doc/source/index.rst

```diff
@@ -62,20 +60,31 @@ How do I get started?
 
 Please visit the :ref:`Overview <overview>` page to learn how to navigate the documentation!
 
+Demo
```

Comment:
<img width="355" alt="Screenshot 2025-05-17 at 4 49 47 PM" src="https://github.com/user-attachments/assets/7506a666-9df1-4a06-94a7-f88559a4fe42" />

I added the demo right below the "Getting started". I initially placed this demo gif at the top but I felt it was  a bit too distracting .